min-stack/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+struct MinStack {
+    stack: Vec<i32>,
+    min_stack: Vec<i32>,
+}
+
+/**
+ * `&self` means the method takes an immutable reference.
+ * If you need a mutable reference, change it to `&mut self` instead.
+ */
+impl MinStack {
+    fn new() -> Self {
+        Self {
+            stack: Vec::new(),
+            min_stack: Vec::new(),
+        }
+    }
+
+    fn push(&mut self, val: i32) {
+        if self.min_stack.is_empty() {
+            self.min_stack.push(val);
+        } else {
+            if self.min_stack[self.min_stack.len() - 1] >= val {
+                self.min_stack.push(val);
+            }
+        }
```

Comment:
Just wondering: could we write the next `if` condition without being nested in the `else` ? or is there a reason why you used `else` here?

Reply:
eg.
```
if self.min_stack.is_empty() {
    self.min_stack.push(val);
}

 if self.min_stack[self.min_stack.len() - 1] >= val {
   self.min_stack.push(val);
}
```